self-tests.yml

```diff
@@ -693,49 +727,58 @@ tasks:
       - func: get-project-and-modules
       - func: verify-merge-function-update
   - name: generate-api-docs
+    patch_only: true
     commands:
       - func: get-project-and-modules
-      - command: subprocess.exec
+      - func: generate-api-docs
+      - func: push-api-docs-task
+  - name: push-api-docs
+    commands:
+      - func: get-project-and-modules
+      - func: generate-api-docs
+      - command: s3.get
         params:
-          binary: make
-          args: ["swaggo-install"]
-          working_dir: evergreen
-          env:
-            GOPATH: ${workdir}/gopath
-          include_expansions_in_env:
-            - GOROOT
-      - command: shell.exec
+          aws_key: ${AWS_ACCESS_KEY_ID}
+          aws_secret: ${AWS_SECRET_ACCESS_KEY}
+          aws_session_token: ${AWS_SESSION_TOKEN}
+          local_file: evergreen/docs/swagger.old.sum
+          remote_file: evergreen/latest/swagger.sum
+          bucket: mciuploads
+          content_type: text/plain
+          optional: true
+      - command: subprocess.exec
         params:
-          shell: bash
           working_dir: evergreen
+          binary: bash
+          args: ["scripts/prepare-swagger-push.sh"]
           env:
-            GOPATH: ${workdir}/gopath
-          script: |
-            set -o verbose
-            set -o errexit
-            $GOPATH/bin/swag init -g service/service.go
+            SWAGGER_JSON_FILE: docs/swagger.json
+            SWAGGER_OLD_SUM_FILE: docs/swagger.old.sum
+            OUTPUT_SUM_FILE: docs/swagger.sum
+      - func: push-api-docs-task
       - command: s3.put
         params:
           aws_key: ${AWS_ACCESS_KEY_ID}
           aws_secret: ${AWS_SECRET_ACCESS_KEY}
           aws_session_token: ${AWS_SESSION_TOKEN}
           local_file: evergreen/docs/swagger.json
-          remote_file: evergreen/${task_id}/${execution}/swagger.json
+          remote_file: evergreen/latest/swagger.json
```

Comment:
Does the `latest` convention still work if these tasks run out of order? Like if commit C1 is pushed, then C2 is pushed, but C2's push task runs before C1's push task, is the `latest` going to be the swagger file from C1 or C2?

Reply:
It's going to be the latest ran (which is how it would operate before). I think in general it's a little weird since it isn't tied to our deploy script, but rather the waterfall commits. I'm going to investigate if we can do this in the drone deploy steps instead so that reverting and inconsistent task timings doesn't result in inconsistent behavior as well.

Reply:
We discussed this in DMs and decided that this should probably be fixed eventually to avoid accidentally overwriting the latest spec file with an older version, but not critical to the ask in the ticket since the existing s3.put for `swagger.json` already had this issue.